stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -496,19 +518,19 @@ for (T, trans, real) in [(:Symmetric, :transpose, :identity), (:(Hermitian{<:Uni
     end
 end
 
-(-)(A::Symmetric) = Symmetric(-A.data, sym_uplo(A.uplo))
-(-)(A::Hermitian) = Hermitian(-A.data, sym_uplo(A.uplo))
+(-)(A::Symmetric) = Symmetric(parentof_applytri(-, A), sym_uplo(A.uplo))
+(-)(A::Hermitian) = Hermitian(parentof_applytri(-, A), sym_uplo(A.uplo))
 
 ## Addition/subtraction
 for f ∈ (:+, :-), (Wrapper, conjugation) ∈ ((:Hermitian, :adjoint), (:Symmetric, :transpose))
     @eval begin
         function $f(A::$Wrapper, B::$Wrapper)
             if A.uplo == B.uplo
-                return $Wrapper($f(parent(A), parent(B)), sym_uplo(A.uplo))
+                return $Wrapper(parentof_applytri($f, A, B), sym_uplo(A.uplo))
             elseif A.uplo == 'U'
-                return $Wrapper($f(parent(A), $conjugation(parent(B))), :U)
+                return $Wrapper(parentof_applytri($f, A, B, identity, $conjugation), :U)
             else
-                return $Wrapper($f($conjugation(parent(A)), parent(B)), :U)
+                return $Wrapper(parentof_applytri($f, A, B, $conjugation, identity), :U)
```

Comment:
These branches have an analogue in `apply_tri`. Couldn't we reduce this to a one-line call, add an internal

```julia
conjugation(::Symmetric) = transpose
conjugation(::Hermitian) = adjoint
```

and include that in `apply_tri`? As an example...

Reply:
Updated